sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,185 @@
+
+## Prerequisites
+
+- [Download the template project](/docs/cheerpj3/tutorials/CheerpJInteroperabilityTutorial.zip) and unzip it.
+- [Node.js](https://nodejs.org/en/) (>= 18)
+
+## Project Structure
+
+The project consists of the following files and directories:
+
+```plaintext
+CheerpJInteroperabilityTutorial/
+├── com/example
+│   └── Example.java
+│   └── manifest.txt
+│   └── Example.class (and other compiled .class files)
+├── index.html
+└── example.jar
+└── Makefile
+
+```
+
+## Java Source Code: `Example.java`
+
+Let's start by examining the source code of the Java class that will interact with JavaScript. This class will receive input from JavaScript, process it, and send a response back to JavaScript.
+
+```java title="Example.java"
+package com.example;
+
+public class Example {
+    public static native void sendToHTML(String s);
+
+    public String processInput(String input) {
+        sendToHTML(input);
+        return "Java received: " + input;
+    }
+
+    public static native void nativeSetApplication(Example myApplication);
+
+    public static void main(String[] args) {
+        Example app = new Example();
+        new Thread(() -> {
+            nativeSetApplication(app);
+            System.out.println("Starting Thread");
+        }).start();
+    }
+}
+```
+
+This Java class contains the following methods:
+
+- `sendToHTML`: A native method implemented in JavaScript, called to send data to the JavaScript environment.
+- `processInput`: Receives data from JavaScript, passes it to `sendToHTML`, and returns a response.
+- `nativeSetApplication`: Captures the running Java thread for persistent communication with JavaScript.
+
+## JavaScript Native Method Implementation
+
+To enable Java and JavaScript communication, we need to implement the [`native`] methods in JavaScript. These methods will handle the data exchange between the two environments.
+
+### 1. Include the CheerpJ Loader Script
+
+Add the following script in the `<head>` section of `index.html`:
+
+```js title="index.html"
+<script src="https://cjrtnc.leaningtech.com/3.0/cj3loader.js"></script>
+```
+
+### 2. Define JavaScript Native Methods
+
+In the `<script>` tag, implement the native methods for interacting with Java.
+
+To implement the native methods in JavaScript, we create an `async` function for each Java method. You can move these methods to a separate `.js` file for better organization and include them using the `<script>` tag in the HTML.
+
+```js title="index.html"
+async function Java_com_example_Example_sendToHTML(lib, str) {
+	document.getElementById("javaOutput").innerText =
+		"JavaScript received: " + str;
+	console.log("Received input from Java: " + str);
+}
+
+async function Java_com_example_Example_nativeSetApplication(
+	lib,
+	myApplication
+) {
+	window.myApplication = myApplication;
+	console.log("Java application instance set on JavaScript side.");
+	return new Promise(() => {}); // Keeps the function from returning
+}
+```
```

Comment:
You should briefly talk about the naming convention and parameters (lib) after showing the implementation.